block/local/adapter.go

```diff
@@ -127,7 +140,28 @@ func (l *Adapter) Remove(obj block.ObjectPointer) error {
 		return err
 	}
 	p = filepath.Clean(p)
-	return os.Remove(p)
+	err = os.Remove(p)
+	if err != nil {
+		return err
+	}
+	if l.removeEmptyDir {
+		dir := filepath.Dir(p)
+		removeEmptyDirUntil(dir, l.path)
+	}
+	return nil
+}
+
+func removeEmptyDirUntil(dir string, stopAt string) {
+	for stopAt != "" && !strings.HasPrefix(stopAt, dir) {
```

Comment:
Sorry, still not sure why the reversal is the safer choice.
```suggestion
	for stopAt != "" && strings.HasPrefix(dir, stopAt) {
```
has the advantage (to my mind) that it stops sooner rather than later when `stopAt` is not a prefix of `dir`.  (You might want to ensure `stopAt` ends in a slash, but that is still safer).  Unless I'm very confused, this code does the wrong thing if an external user can convince a local adapter to remove a file that is not under the control of the adapter.  This is currently *NOT* possible, but I still prefer the safer code.

Reply:
To make `stopAt != "" && strings.HasPrefix(dir, stopAt)` work I also need the add the == case.
In case we use "local://" I don't want to delete the adapter directory.

Reply:
Updated the code